test/statistics.jl

```diff
@@ -26,6 +26,19 @@
     @test quants2D[2][:value] == quant2D[:value]
   end
 
+  @testset "HalfSampleMode" begin
+    d = LogNormal(0,1)
+    rs = rand(d,1000)
+    @test GeoStatsBase.hsm_mode([1,2,2,3]) == 2.0
+    @test GeoStatsBase.hsm_mode([1,2,2,3,5]) == 2.0
+    @test GeoStatsBase.hsm_mode(rs) < mean(rs)
+    @test GeoStatsBase.hsm_mode(rs) < median(rs)
+    d = Normal(1,0.5)
+    rs = vcat(rand(Normal(0,1), 700), rand(Normal(3,0.2), 300))
```

Comment:
Should we rewrite this as `d = MixtureModel([Normal(), Normal(3, 0.2)], [0.7, 0.3])` and then `rand(d, 1000)`?